lib/markout/construct-videos.js

```diff
@@ -14,14 +14,21 @@ function constructVideo($) {
         var type = mediaType("video", src);
         
         if (type) {
-            if(type === "link"){
+            if(type === "link") {
                 var regex = RegExp("(youtube.com|youtu.be)");
                 if (regex.test(src)){ // Check if it's youtube link
-                    var youtubeId = src.split("&")[0];
-                    youtubeId = youtubeId.substr(youtubeId.length - 11);
+                    var start_index = src.indexOf("v=");
```

Comment:
@Arv91 @kphunter I would suggest adopting a query-string parsing library.  It will be less error prone, easier to use, and easier to maintain.  I will make other enhancements less daunting too.